### PR TITLE
[RWE-9044]][risk=no] Add logic to determine whether to reuse the same disk or creating new disk when creating app

### PR DESCRIPTION
**Pull request by yonghaoy**
Description:

- This is the same logic as Runtime creation: Frontend will pass PD name if PD exists. Otherwise pass empty name.
- Starting setting labels to both RUNTIME and APP(set app type)
- Also, this PR refactor the code, and extra common method from runtime

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have added explanatory comments where the logic is not obvious
- [ ] I have run and tested this change locally, and my testing process is described here
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later
- [ ] If this includes an API change, I have run the E2E tests on this change against my local server with [yarn test-local](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples) because this PR won't be covered by the CircleCI tests 
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P)
- [ ] If this change impacts deployment safety (e.g. removing/altering APIs which are in use) I have documented these in the description
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and updated the appropriate API consumers
  * AoU UI
  * [Perf tests](https://github.com/broadinstitute/mcnulty/blob/develop/src/test/scala/services/AoU.scala)
  * [Researcher Directory export](https://github.com/all-of-us/workbench/wiki/Researcher-Directory-(RDR-export))
  * Cron tasks - for Offline*Controllers
  * SumoLogic - for EgressAlert

### CODE REVIEW

File: api/src/test/java/org/pmiops/workbench/leonardo/LeonardoLabelHelperTest.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package org.pmiops.workbench.leonardo;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.jupiter.api.Test;
+
+public class LeonardoLabelHelperTest {
```
Comment (Peter-Lavigne) [inline api/src/test/java/org/pmiops/workbench/leonardo/LeonardoLabelHelperTest.java:9]: Should `appTypeToLabelValue` also be tested? Or do you consider it tested by the API client tests?

File: ui/public/index.html [hunk 2]
```diff
@@ -10,7 +10,7 @@
     <!-- Google Platform JS for authentication
       This puts `gapi` into the global namespace
     -->
-    <script src="https://apis.google.com/js/platform.js"></script>
+    <script src="https://accounts.google.com/gsi/client" async defer></script>
```
Comment (Peter-Lavigne) [inline ui/public/index.html:13]: What is this change doing?

File: api/src/main/java/org/pmiops/workbench/leonardo/LeonardoApiClientImpl.java [hunk 3]
```diff
@@ -505,26 +510,39 @@ public List<LeonardoListPersistentDiskResponse> listPersistentDiskByProject(
...
+            .toLeonardoPersistentDiskRequest(createAppRequest.getPersistentDiskRequest())
+            .labels(
+                upsertLeonardoLabel(
+                    createAppRequest.getPersistentDiskRequest().getLabels(),
+                    LEONARDO_LABEL_APP_TYPE,
+                    appTypeToLabelValue(createAppRequest.getAppType())));
+    // If diskRequest is not set or missing name from request, that means creating new disk.
+    if (diskRequest != null && Strings.isNullOrEmpty(diskRequest.getName())) {
+      diskRequest.setName(userProvider.get().generatePDNameForApp(appType));
+    }
```
Comment (Peter-Lavigne) [inline api/src/main/java/org/pmiops/workbench/leonardo/LeonardoApiClientImpl.java:534]: Product question: Do all apps have PDs?

Are `If diskRequest is not set` and `diskRequest != null` saying different things?

Comment (yonghaoy) [reply api/src/main/java/org/pmiops/workbench/leonardo/LeonardoApiClientImpl.java:534]: If diskRequest is not set is the same as `diskRequest != null`
or missing name from request is `Strings.isNullOrEmpty(diskRequest.getName())`

File: ui/public/index.html (hunk 2 above)
Comment (yonghaoy) [reply ui/public/index.html:13]: ah, sorry, I was trying to size https://precisionmedicineinitiative.atlassian.net/browse/RW-8907, did not realize I was using this branch. Reverted.

File: api/src/test/java/org/pmiops/workbench/leonardo/LeonardoLabelHelperTest.java (hunk 1 above)
Comment (yonghaoy) [reply api/src/test/java/org/pmiops/workbench/leonardo/LeonardoLabelHelperTest.java:9]: added a new tests. PTAL
Original that was covered in LeoApiTest, and I feel that is too simple to add test. 
But couldn't hurt to add simple test

File: api/src/main/java/org/pmiops/workbench/leonardo/LeonardoApiClientImpl.java (hunk 3 above)
Comment (yonghaoy) [reply api/src/main/java/org/pmiops/workbench/leonardo/LeonardoApiClientImpl.java]: update: Remove diskRequest null check

**Review by Peter-Lavigne — APPROVED** [review]

**Pull request merged**